lib/bootstrap.js

```diff
@@ -0,0 +1,95 @@
+import UiAutomator from 'appium-uiautomator';
+import ADB from 'appium-adb';
+import log from './logger';
+import net from 'net';
+import path from 'path';
+
+class AndroidBootstrap {
+  constructor (systemPort, webSocket) {
+    this.systemPort = systemPort;
+    this.webSocket = webSocket;
+  }
+
+  async start (appPackage, disableAndroidWatchers) {
+    try {
+      let rootDir = path.resolve(__dirname, '../..');
+      let startDetector = (s) => { return /Appium Socket Server Ready/.test(s); };
+      const bootstrapJar = path.resolve(rootDir, 'bootstrap', 'bin', 'AppiumBootstrap.jar');
+      await this.createUiAutomator();
+      this.adb.forwardPort(this.systemPort, this.systemPort);
+      this.process = await this.uiAutomator.start(
+                       bootstrapJar, 'io.appium.android.bootstrap.Bootstrap',
+                       startDetector, '-e', 'pkg', appPackage, '-e',
+                       'disableAndroidWatchers', disableAndroidWatchers);
+      this.process.on('output', (stdout, stderr) => {
+        const alertRe = /Emitting system alert message/;
+        if (alertRe.test(stdout)) {
+          log.debug("Emitting alert message...");
+          this.webSocket.sockets.emit('alert', {message: stdout});
+        }
+        log.debug(`stdout: ${stdout}`);
+        log.debug(`stderr: ${stderr}`);
+      });
+
+      this.socketClient = net.connect(this.systemPort, () => {
+        log.info("Connected");
+      });
+    } catch (e) {
+      log.errorAndThrow(`Error occured while starting AndroidBootstrap. Original error: ${e}`);
+    }
+  }
+
+  sendCommand (type, extra = {}) {
+    return new Promise ((resolve, reject) => {
+      if (this.socketClient) {
+        setTimeout(() => {
+          reject(new Error("Bootstrap server did not respond in 10000"));
+        }, 10000);
+        let cmd = {cmd: type};
+        cmd = Object.assign(cmd, extra);
+        let cmdJson = JSON.stringify(cmd) + "\n";
+        let logCmd = cmdJson.trim();
+        if (logCmd.length > 1000) {
```

Comment:
This logic should probably be moved into `appium-support`. I also have something similar [here](https://github.com/appium/appium-remote-debugger/blob/master/lib/helpers.js#L128).